### PR TITLE
Add matchUnescapedFirst to routing doc

### PR DESCRIPTION
**Pull request by ctgowrie**
Adds a new field to routing API to allow optionally matching on an unescaped path first

### CODE REVIEW

**Review by ijjk — COMMENTED** [review]
We also need to update the route type here to allow it https://github.com/vercel/vercel/blob/326fe0f0e647a80ef504ca4035c6bc3c206f43d6/packages/routing-utils/src/types.ts#L53

File: packages/routing-utils/src/schemas.ts [hunk 1]
```diff
@@ -116,6 +116,9 @@ export const routesSchema = {
           isInternal: {
             type: 'boolean',
           },
+          matchUnescapedFirst: {
```
Comment (ijjk) [inline packages/routing-utils/src/schemas.ts:119]: Should we shorten this to just `unescape: true | undefined` or would that be too vague on what's being unescaped?

Comment (EndangeredMassa) [reply packages/routing-utils/src/schemas.ts:119]: The longer name makes more sense to me

**Review by EndangeredMassa — APPROVED** [review]

**Review by ijjk — APPROVED** [review]
